File: pbx-platform/services/api/app/schemas/consult.py
from datetime import datetime
from typing import Optional, List
from pydantic import BaseModel

from pbx_common.models.consultation import ConsultationStatus

class AgentBrief(BaseModel):
    id: int
    name: str
    class Config:
        from_attrbutes = True

class CompanyBrief(BaseModel):
    id: int
    company_name: str
    class Config:
        from_attributes = True

class CategoryBrief(BaseModel):
    id: int
    name: str
    depth: int
    class Config:
        from_attributes = True

class ConsultationCreate(BaseModel):
    agent_id: int
    company_id: int
    call_id: Optional[str] = None
    category_id: Optional[int] = None
    category: Optional[str] = None
    memo: Optional[str] = None
    started_at: Optional[datetime] = None
    ended_at: Optional[datetime] = None

class ConsultationUpdate(BaseModel):
    memo: Optional[str] = None
    category: Optional[str] = None
    category_id: Optional[int] = None
    ended_at: Optional[datetime] = None

class ConsultationLinkCall(BaseModel):
    call_id: str

class ConsultationResponse(BaseModel):
    id: int
    call_id: Optional[str] = None
    agent_id: int
    company_id: int
    original_id: Optional[int] = None
    category_id: Optional[int] = None
    category: Optional[str] = None
    memo: Optional[str] = None
    status: ConsultationStatus
    started_at: Optional[datetime] = None
    ended_at: Optional[datetime] = None
    created_at: datetime
    updated_at: datetime
    agent: Optional[AgentBrief] = None
    company: Optional[CompanyBrief] = None
    category_obj: Optional[CategoryBrief] = None

    class Config:
        from_attributes = True

class ConsultCategoryCreate(BaseModel):
    company_id: int
    parent_id: Optional[int] = None
    name: str
    depth: int
    sort_order: int = 0

class ConsultCategoryUpdate(BaseModel):
    name: Optional[str] = None
    sort_order: Optional[int] = None
    is_active: Optional[bool] = None

class ConsultCategoryResponse(BaseModel):
    id: int
    company_id: int
    parent_id: Optional[int] = None
    name: str
    depth: int
    sort_order: int
    is_active: bool
    created_at: datetime
    children: List["ConsultCategoryResponse"] = []

    class Config:
        from_attributes = True

ConsultCategoryResponse.model_rebuild()

File: pbx-platform/services/frontend/types/consult.ts
export interface ConsultAgent {
    id: number;
    name: string;
}

export interface ConsultCompany {
    id: number;
    company_name: string;
}

export interface ConsultCategoryBrief {
    id: number;
    name: string;
    depth: number;
}

export interface ConsultCategory {
    id: number;
    company_id: number;
    parent_id: number | null;
    name: string;
    depth: number;
    sort_order: number;
    is_active: boolean;
    created_at: string;
    children: ConsultCategory[];
}

export interface Consultation {
    id: number;
    call_id: string | null;
    agent_id: number;
    company_id: number;
    original_id: number | null;
    category_id: number | null;
    memo: string | null;
    status: "ACTIVE" | "INACTIVE" | "SUPERSEDED";
    started_at: string | null;
    ended_at: string | null;
    created_at: string;
    updated_at: string;
    agent: ConsultAgent | null;
    company: ConsultCompany | null;
    category_obj: ConsultCategoryBrief | null;
}